src/bosh_aws_cpi/lib/cloud/aws/cloud.rb

```diff
@@ -346,19 +346,27 @@ def snapshot_disk(disk_id, metadata)
         volume.attachments.each { |attachment| devices << attachment.device }
 
         name = ['deployment', 'job', 'index'].collect { |key| metadata[key] }
-        name << devices.first.split('/').last unless devices.empty?
+
+        unless devices.empty?
+          name << devices.first.split('/').last
+          metadata['device'] = devices.first
+        end
 
         snapshot = volume.create_snapshot(name.join('/'))
         logger.info("snapshot '#{snapshot.id}' of volume '#{disk_id}' created")
 
+        metadata.merge!({
+          'director' => metadata['director_name'],
+          'instance_index' => metadata['index'].to_s,
+          'instance_name' => metadata['job'] + '/' + metadata['instance_id'],
+          'Name' => name.join('/')
+        })
 
-        tags = {}
-        ['agent_id', 'instance_id', 'director_name', 'director_uuid'].each do |key|
-          tags[key] = metadata[key]
+        ['director_name', 'index', 'job'].each do |tag|
```

Comment:
Is there a reason we specifically need to delete these keys? I'd expect CPI to be fairly oblivious, and if the director wants them, they should be tagged. Rather, I'd expect CPIs to be additive with more keys if relevant to CPI, but not to delete them, in general. Maybe related to above question.